buildtest/cli/buildspec.py

```diff
@@ -882,7 +882,9 @@ def print_invalid_buildspecs(self, error=None, terse=None, header=None):
                 print("buildspec")
             for buildspec in self.cache["invalids"].keys():
                 print(buildspec)
-            sys.exit(1)
+            if self.cache["invalids"]:
+                sys.exit(1)
```

Comment:
so i think we can refactor this codebase now that i think about it you can call at beginning of method  after these lines

```
        if error and terse:
            console.print("The --terse flag can not be used with the --error option")
            sys.exit(1)
```

The following check for any invalid buildspecs and print a message and just return. Something like this

```
if not self.get_invalid_buildspecs():
  console.print("There are no invalid buildspecs in cache")
  return
```

Now after this statement we know any subsequent command will imply there is an invalid buildspec in cache and we can call `sys.exit(1)` at end of method instead of doing a conditional check. Can you please refactor this method i hope i make myself clear.

Reply:
Yes Right! I have refactored it. Also, I guess we donot require to check for the number of rows in the table in error=false condition so I have remove it.
`   # return a non-zero returncode
            if table.row_count > 0:
                console.print(table)
                sys.exit(1)
            return`